util.lua

```diff
@@ -0,0 +1,40 @@
+-- Copyright (c) 2021 EngineerSmith
+-- Under the MIT license, see license suppiled with this file
+
+local util = {}
+
+util.getImageDimensions = function(image)
+  if image:typeOf("Texture") then
+    return image:getPixelDimensions()
+  else
+    return image:getDimensions()
+  end
+end
+
+
+-- TODO: Remove closure usage
+local fillImageData = function(imageData, x, y, w, h, r, g, b, a)
+  imageData:mapPixel(function()
+    return r, g, b, a
+  end, x, y, w, h)
+end
+
+local extrudeImageData = function(dest, src, n, x, y, dx, dy, sx, sy, sw, sh)
+  for i = 1, n do
+    dest:paste(src, x + i * dx, y + i * dy, sx, sy, sw, sh)
+  end
+end
+
+util.extrudeWithFill = function(dest, src, n, x, y)
+  local iw, ih = src:getDimensions()
+  extrudeImageData(dest, src, n, x, y, 0, -1, 0, 0, iw, 1) -- top
+  extrudeImageData(dest, src, n, x, y, -1, 0, 0, 0, 1, ih) -- left
+  extrudeImageData(dest, src, n, x, y + ih - 1, 0, 1, 0, ih - 1, iw, 1) -- bottom
+  extrudeImageData(dest, src, n, x + iw - 1, y, 1, 0, iw - 1, 0, 1, ih) -- right
+  fillImageData(dest, x - n - 1, y - n - 1, n, n, src:getPixel(0, 0)) -- top-left
```

Comment:
Top-left, Top-right and Bottom-left are misalligned for both fixedSized and dynamicSized
1 extrude ![image](https://user-images.githubusercontent.com/56016593/151394293-21fddcd4-00b4-42d4-933f-23f05388eacc.png)
2 extrude ![image](https://user-images.githubusercontent.com/56016593/151394478-da144add-720f-4293-b5e8-8e20f327fb17.png)